apis/eventing/v1alpha1/cloudeventsource_webhook.go

```diff
@@ -0,0 +1,102 @@
+
+var cloudeventsourcelog = logf.Log.WithName("cloudeventsource-validation-webhook")
+
+func (ces *CloudEventSource) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(ces).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/validate-eventing-keda-sh-v1alpha1-cloudeventsource,mutating=false,failurePolicy=ignore,sideEffects=None,groups=eventing.keda.sh,resources=cloudeventsources,verbs=create;update,versions=v1alpha1,name=vcloudeventsource.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &CloudEventSource{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (ces *CloudEventSource) ValidateCreate() (admission.Warnings, error) {
+	val, _ := json.MarshalIndent(ces, "", "  ")
+	cloudeventsourcelog.Info(fmt.Sprintf("validating cloudeventsource creation for %s", string(val)))
+	return validateSpec(&ces.Spec)
+}
+
+func (ces *CloudEventSource) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
+	val, _ := json.MarshalIndent(ces, "", "  ")
+	cloudeventsourcelog.V(1).Info(fmt.Sprintf("validating cloudeventsource update for %s", string(val)))
+
+	oldCes := old.(*CloudEventSource)
+	if isCloudEventSourceRemovingFinalizer(ces.ObjectMeta, oldCes.ObjectMeta, ces.Spec, oldCes.Spec) {
+		cloudeventsourcelog.V(1).Info("finalizer removal, skipping validation")
+		return nil, nil
+	}
+	return validateSpec(&ces.Spec)
+}
+
+func (ces *CloudEventSource) ValidateDelete() (admission.Warnings, error) {
+	return nil, nil
+}
+
+func isCloudEventSourceRemovingFinalizer(om metav1.ObjectMeta, oldOm metav1.ObjectMeta, spec CloudEventSourceSpec, oldSpec CloudEventSourceSpec) bool {
+	cesSpec, _ := json.MarshalIndent(spec, "", "  ")
+	oldCesSpec, _ := json.MarshalIndent(oldSpec, "", "  ")
+	cesSpecString := string(cesSpec)
+	oldCesSpecString := string(oldCesSpec)
+
+	return len(om.Finalizers) == 0 && len(oldOm.Finalizers) == 1 && cesSpecString == oldCesSpecString
+}
+
+func validateSpec(spec *CloudEventSourceSpec) (admission.Warnings, error) {
+	if spec.EventSubscription.ExcludedEventTypes != nil {
+		for _, excludedEventType := range spec.EventSubscription.ExcludedEventTypes {
+			if !slices.Contains(AllEventTypes, excludedEventType) {
+				return nil, fmt.Errorf("excludedEventType: %s in cloudeventsource spec is not supported", excludedEventType)
+			}
+		}
+	}
+
+	if spec.EventSubscription.IncludedEventTypes != nil {
+		for _, includedEventType := range spec.EventSubscription.IncludedEventTypes {
+			if !slices.Contains(AllEventTypes, includedEventType) {
+				return nil, fmt.Errorf("includedEventType: %s in cloudeventsource spec is not supported", includedEventType)
+			}
+		}
+	}
+
+	if spec.EventSubscription.ExcludedEventTypes != nil && spec.EventSubscription.IncludedEventTypes != nil {
+		for _, excludedEventType := range spec.EventSubscription.ExcludedEventTypes {
+			if slices.Contains(spec.EventSubscription.ExcludedEventTypes, excludedEventType) {
```

Comment:
I think that you want to check the excluded event within included events
```suggestion
			if slices.Contains(spec.EventSubscription.IncludedEventTypes, excludedEventType) {
```